src/HookManager.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#include "HookManager.h"
+
+#include <QRegExp>
+
+#include "HookFormatter.h"
+
+HookManager::HookManager(QObject* parent)
+    : QObject(parent),
+    m_event(),
+    m_eventSpec(),
+    m_hooksSignedUp()
+{
+}
+
+
+void HookManager::connectHook(QByteArray eventSpec, HookFormatter* formatter)
+{
+    /// Remove keyword before key sequence, indicated by colon. At the moment,
+    /// everything up to (and including) the colon is removed and discarded, 
+    /// so not specifying 'key:' works as well.
+    QString specifier = QString::fromUtf8(eventSpec);
+    //specifier.count(QRegExp("[:]"));
+    specifier.replace(QRegExp(".*[:]"), "");
```

Comment:
I think we should be prescriptive here and only accept a `"key:"` prefix.  Otherwise people will come to rely on the permissive API and their scripts will break later on.